src/main/java/spireMapOverhaul/zones/brokenSpace/DrawCardWithCallbackAction.java

```diff
@@ -0,0 +1,155 @@
+//
+// Source code recreated from a .class file by IntelliJ IDEA
+// (powered by FernFlower decompiler)
+//
+
+package spireMapOverhaul.zones.brokenSpace;
+
+import com.badlogic.gdx.Gdx;
+import com.megacrit.cardcrawl.actions.AbstractGameAction;
+import com.megacrit.cardcrawl.actions.AbstractGameAction.ActionType;
+import com.megacrit.cardcrawl.actions.common.EmptyDeckShuffleAction;
+import com.megacrit.cardcrawl.cards.AbstractCard;
+import com.megacrit.cardcrawl.cards.SoulGroup;
+import com.megacrit.cardcrawl.core.AbstractCreature;
+import com.megacrit.cardcrawl.core.Settings;
+import com.megacrit.cardcrawl.dungeons.AbstractDungeon;
+import com.megacrit.cardcrawl.vfx.PlayerTurnEffect;
+
+import java.util.ArrayList;
+import java.util.function.Consumer;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+public class DrawCardWithCallbackAction extends AbstractGameAction {
```

Comment:
Unneeded, DrawCardAction can already do this, check out Scrape for an example. Also, this class checks for a harcoded handsize of 10 when it should check for BaseMod.MAX_HAND_SIZE. This is why you should always try to avoid copy pasting decompiled classes.